docs/raw/serving-community-history.md

```diff
@@ -0,0 +1,406 @@
+
+### WakuMessageArchiveIndex
+
+A `WakuMessageArchiveIndex` is a map where the key is the KECCAK-256 hash of the `WakuMessageArchiveIndexMetadata` derived from a 7-day archive and the value is an instance of that `WakuMessageArchiveIndexMetadata` corresponding to that archive.
+
+The `offset` field MUST contain the position at which the message history archive starts in the byte string of the total message archive data. This MUST be the sum of the length of all previously created message archives in bytes (see [Creating message archive torrents](#creating-message-archive-torrents)).
+
+```
+syntax = "proto3"
+
+message WakuMessageArchiveIndexMetadata {
+  uint8 version = 1
+  WakuMessageArchiveMetadata metadata = 2
+  uint64 offset = 3
+  uint64 num_pieces = 4
+}
+
+message WakuMessageArchiveIndex {
+  map<string, WakuMessageArchiveIndexMetadata> archives = 1
+}
+```
+
+The community owner node MUST update the `WakuMessageArchiveIndex` every time it creates one or more `WakuMessageArchive`s and bundle it into a new torrent (**TODO: see section**).
+For every created `WakuMessageArchive`, there MUST be a `WakuMessageArchiveIndexMetadata` entry in the `archives` field `WakuMessageArchiveIndex`.
+
+## Creating message archive torrents
+
+Community owner nodes MUST create a torrent file ("torrent") containing metadata to all message history archives. To create a torrent file, and later serve the message archive data in the BitTorrent network, community owner nodes MUST store the necessary data in dedicated files on the file system.
+
+A torrent's source folder MUST contain the following two files:
+
+- `data` - Contains all protobuf encoded message history archives concatenated in ascending order
+- `index` - Contains the protobuf encoded message history archive index
+
+Community owner nodes SHOULD store these files in a dedicated folder that is identifiable via the community id.
+
+### Ensuring reproducible data pieces
+
+The community owner node MUST ensure that the byte string resulting from the protobuf encoded `data` is equal to the byte string `data` from the previously generated message archive torrent, plus the data of the latest 7 days worth of messages encoded as `WakuMessageArchive`. Therefore, the size of `data` grows every seven days as it's append only.
+
+The community owner nodes also MUST ensure that the byte size of every individual `WakuMessageArchive` encoded protobuf is a multiple of `pieceLength: ???` (**TODO**) using the `padding` field. If the last piece of a message archive has fewer bytes than `pieceLength`, it MUST be filled with zero bytes until it has the size `pieceLength`.
+
+This is necessary because message history archive data will be split into pieces of `pieceLength` when the torrent file is created, and the SHA1 hash of every piece is then stored in the torrent file and later used by other nodes to request the data for each individual data piece.
```

Comment:
```suggestion
This is necessary because the content of the `data` file will be split into pieces of `pieceLength` when the torrent file is created, and the SHA1 hash of every piece is then stored in the torrent file and later used by other nodes to request the data for each individual data piece.
```